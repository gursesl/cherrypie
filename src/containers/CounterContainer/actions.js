import * as c from './constants'

export function incrementAction() {
  return {
    type: c.INCREMENT_ACTION,
  }
}

export function decrementAction() {
  return {
    type: c.DECREMENT_ACTION,
  }
}

export function incrementActionAsync() {
  return {
    type: c.INCREMENT_ACTION_ASYNC,
  }
}
