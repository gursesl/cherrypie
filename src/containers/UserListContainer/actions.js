import * as c from './constants'

export function usersFetchStart() {
  return { type: c.USERS_FETCH_START }
}

export function usersFetchSuccess(users) {
  return {
    type: c.USERS_FETCH_SUCCESS,
    payload: users,
  }
}

export function usersFetchFailure(err) {
  return {
    type: c.USERS_FETCH_FAILURE,
    error: err,
  }
}
