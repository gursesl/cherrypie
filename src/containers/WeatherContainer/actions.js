import * as c from './constants'

export function weatherDataFetchStart() {
  return { type: c.WEATHER_DATA_FETCH_START }
}

export function weatherDataFetchSuccess(cities) {
  return {
    type: c.WEATHER_DATA_FETCH_SUCCESS,
    payload: cities,
  }
}

export function weatherDataFetchFailure(err) {
  return {
    type: c.WEATHER_DATA_FETCH_FAILURE,
    error: err,
  }
}
