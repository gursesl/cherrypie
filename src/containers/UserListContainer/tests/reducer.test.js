import { fromJS } from 'immutable'
import userListReducer from '../reducer'
import * as a from '../actions'

let state
const mockUsers = [
  {
    id: "95617189",
    firstName: "Elfrieda",
    lastName: "Frank",
    email: "Ada20@hotmail.com"
  },
  {
    id: "95617188",
    firstName: "Jim",
    lastName: "Smith",
    email: "jsmith@mail.com"
  },
]
const mockErr = {type: 1, message: "An error occurred."}

describe('userListReducer', () => {

  beforeEach(() => {
    state = fromJS({
      users: [],
      error: null
    })
  })

  it('sould return the initial state', () => {
    expect(userListReducer(undefined, {})).toEqual(state)
  })

  it('should handle usersFetchStart action correctly', () => {
    expect(userListReducer(state, a.usersFetchStart())).toEqual(state)
  })

  it('should handle usersFetchSuccess action correctly', () => {
    const expectedState = fromJS({
      users: mockUsers,
      error: null
    })
    expect(userListReducer(state, a.usersFetchSuccess(mockUsers))).toEqual(expectedState)
  })

  it('should handle usersFetchFailure action correctly', () => {
    const expectedState = fromJS({
      users: [],
      error: mockErr
    })
    expect(userListReducer(state, a.usersFetchFailure(mockErr))).toEqual(expectedState)
  })

})
