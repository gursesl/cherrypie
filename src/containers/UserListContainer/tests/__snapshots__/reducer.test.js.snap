// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListContainer:reducer handles the usersFetchFailure action snapshot 1`] = `
Immutable.Map {
  "app/UserListContainer/SELECTOR_USERS_USERS": Immutable.List [
    Immutable.Map {
      "id": 95617189,
      "firstName": "Elfrieda",
      "lastName": "Frank",
      "email": "Ada20@hotmail.com",
    },
    Immutable.Map {
      "id": 95617188,
      "firstName": "Jim",
      "lastName": "Smith",
      "email": "jsmith@mail.com",
    },
  ],
  "app/UserListContainer/SELECTOR_USERS_ERROR": Immutable.Map {
    "type": 1,
    "message": "An error occurred.",
  },
  "app/UserListContainer/SELECTOR_USERS_ISLOADING": true,
}
`;

exports[`UserListContainer:reducer handles the usersFetchStart action snapshot 1`] = `Object {}`;

exports[`UserListContainer:reducer handles the usersFetchSuccess action snapshot 1`] = `
Immutable.Map {
  "app/UserListContainer/SELECTOR_USERS_USERS": Immutable.List [
    Immutable.Map {
      "id": 95617189,
      "firstName": "Elfrieda",
      "lastName": "Frank",
      "email": "Ada20@hotmail.com",
    },
    Immutable.Map {
      "id": 95617188,
      "firstName": "Jim",
      "lastName": "Smith",
      "email": "jsmith@mail.com",
    },
  ],
  "app/UserListContainer/SELECTOR_USERS_ERROR": "This is an error",
  "app/UserListContainer/SELECTOR_USERS_ISLOADING": true,
}
`;
